### PR TITLE
Eliminate the duplication code of `visit_TableDefinition`

### PR DESCRIPTION
**Pull request by kamipo**

### CODE REVIEW

File: activerecord/lib/active_record/connection_adapters/abstract/schema_creation.rb [hunk 1]
```diff
@@ -93,6 +101,10 @@ def options_include_default?(options)
             options.include?(:default) && !(options[:null] == false && options[:default].nil?)
           end
 
+          def index_in_create(table_name, column_name, options)
```
Comment (yui-knk) [inline activerecord/lib/active_record/connection_adapters/abstract/schema_creation.rb]: I'm not sure `index_in_create` is an interface of abstract adapter.
If we define `index_in_create` as interface, how about add these comments to this method?

> If the adapter's `supports_indexes_in_create` returns +true+, the adapter should implement + index_in_create+ method.

Comment (yui-knk) [inline activerecord/lib/active_record/connection_adapters/abstract/schema_creation.rb]: I think defining `index_in_create` as an interface of abstract adapter is not good. `index_in_create` is only defined on `abstract_mysql_adapter.rb` so I feel it is not an interface.

Comment (sgrif) [inline activerecord/lib/active_record/connection_adapters/abstract/schema_creation.rb]: I agree, this serves no purpose.

Comment (kamipo) [inline activerecord/lib/active_record/connection_adapters/abstract/schema_creation.rb]: I removed this method.

**Pull request merged**